server/services/recommendedVersionsService.ts

```diff
@@ -0,0 +1,208 @@
+ * If not present, it falls back to:
+ *  - helm_deploy/values.yaml for helm dependency versions
+ *  - gradle/libs.versions.toml for hmpps_gradle_spring_boot version
+ *
+ * Results are cached in-memory with TTL to avoid excessive network calls.
+ */
+export default class RecommendedVersionsService {
+  private static cache: { value: RecommendedVersions; expiresAt: number } | null = null
+
+  private serviceCatalogueService?: ServiceCatalogueService
+
+  constructor(
+    private readonly ttlMillis: number = Number(process.env.RECOMMENDED_VERSIONS_TTL_MS) || 6 * 60 * 60 * 1000, // 6h
+  ) {}
+
+  setServiceCatalogueService(service: ServiceCatalogueService): void {
+    this.serviceCatalogueService = service
+  }
+
+  async getRecommendedVersions(): Promise<RecommendedVersions> {
+    const now = Date.now()
+    if (RecommendedVersionsService.cache && RecommendedVersionsService.cache.expiresAt > now) {
+      return RecommendedVersionsService.cache.value
+    }
+
+    const result: RecommendedVersions = {
+      helm_dependencies: {},
+      gradle: {},
+      metadata: { source: 'none', fetchedAt: new Date(now).toISOString() },
+    }
+
+    // Only Strapi: hmpps-template-kotlin component
+    const fromStrapi = await this.tryFetchFromStrapi()
+    logger.info(`[RecommendedVersions] Strapi result: ${JSON.stringify(fromStrapi)}`)
+    if (fromStrapi) {
+      result.helm_dependencies.generic_prometheus_alerts = fromStrapi.helm_generic_prometheus_alerts
+      result.helm_dependencies.generic_service = fromStrapi.helm_generic_service
+      result.gradle.hmpps_gradle_spring_boot = fromStrapi.gradle_hmpps_gradle_spring_boot
+      result.metadata.source = 'strapi'
+    }
+
+    // Mark partial if any key is missing
+    if (
+      !result.helm_dependencies.generic_prometheus_alerts ||
+      !result.helm_dependencies.generic_service ||
+      !result.gradle.hmpps_gradle_spring_boot
+    ) {
+      result.metadata.source = result.metadata.source === 'none' ? 'none' : 'partial'
+    }
+
+    RecommendedVersionsService.cache = {
+      value: result,
+      expiresAt: now + this.ttlMillis,
+    }
+
+    logger.info(
+      `[RecommendedVersions] Source=${result.metadata.source}, helm(gpa=${
+        result.helm_dependencies.generic_prometheus_alerts || 'missing'
+      }, gs=${result.helm_dependencies.generic_service || 'missing'}), gradle(hgsb=${
+        result.gradle.hmpps_gradle_spring_boot || 'missing'
+      })`,
+    )
+
+    return result
+  }
+
+  private parseVersionValue(raw: unknown): string | undefined {
+    if (raw === null || raw === undefined) return undefined
+    if (typeof raw === 'string' || typeof raw === 'number') return String(raw)
+    if (typeof raw === 'object') {
+      const obj = raw as Record<string, unknown>
+      return (obj.ref as string) || (obj.version as string) || undefined
+    }
+    return undefined
+  }
+
+  private async tryFetchFromStrapi(): Promise<{
```

Comment:
Function name needs an actual name, not sure what it's supposed to be doing from the current name?